### PR TITLE
CLN, TYP Remove string type hints

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

For some background: there was some consensus on doing this [here](https://github.com/pandas-dev/pandas/pull/36034#issuecomment-685186742)

Then [here](https://github.com/pandas-dev/pandas/pull/39270#pullrequestreview-571190631) there was the suggestion to use `ast.parse`, rather than regular expressions. I had a go at that, and made this into its own package: https://github.com/MarcoGorelli/no-string-hints , which I've included here as a pre-commit hook

The changes here are just a result of running `pre-commit run no-string-hints -a`, and then adding `from __future__ import annotations` to some files.

### CODE REVIEW

**Review by jreback — APPROVED** [review]
wow. lgtm. cc @simonjayhawkins 

just have to make sure the author sticks around :->

File: pandas/core/internals/construction.py [hunk 1]
```diff
@@ -65,8 +67,6 @@
 if TYPE_CHECKING:
     from numpy.ma.mrecords import MaskedRecords
 
-    from pandas import Series
```
Comment (rhshadrach) [inline pandas/core/internals/construction.py:68]: Curious what happened here. Series is imported inline below, so am I correct to say that this is sufficient for mypy?

**Review by rhshadrach — COMMENTED** [review]
This is great! Just one question below.

File: pandas/core/internals/construction.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/core/internals/construction.py:68]: Thanks!

And yes, seems like it, e.g.:
`t.py`:
```python
def foo() -> None:
    a: List[int]
    reveal_type(a)
    from typing import List
```

```console
$ python3.8 -m mypy t.py 
t.py:3: note: Revealed type is 'builtins.list[builtins.int]'
```

**Comment by simonjayhawkins** [comment]
Thanks @MarcoGorelli. this is great.

**Comment by MarcoGorelli** [comment]
Thanks all for your kind words, much appreciated :pray: !

**Pull request merged**